From 16c4155655e63d5e5ee1e463b6e29038870c68a8 Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anuj.mittal@intel.com>
Date: Wed, 29 Nov 2023 12:42:57 +0530
Subject: [PATCH] cmake: yocto specific tweaks to the build process

* Dont try to detect glibc version as that doesn't work when cross compiling.
* Dont try to detect CXX11_ABI
* Install sample binaries as well.
* Dont try to write triggers for CPack. We package ourselves.
* Fix the installation path for Python modules when baselib = lib64.
* python3-config is installed as python3-config in oe-core Python recipe

Upstream-Status: Inappropriate

Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>
---
 cmake/developer_package/cross_compile/python_helpers.cmake | 2 +-
 cmake/developer_package/packaging/rpm/rpm.cmake            | 2 +-
 cmake/developer_package/target_flags.cmake                 | 4 ++--
 samples/cpp/CMakeLists.txt                                 | 6 +++---
 src/bindings/python/CMakeLists.txt                         | 2 +-
 5 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/cmake/developer_package/cross_compile/python_helpers.cmake b/cmake/developer_package/cross_compile/python_helpers.cmake
index a6c7de73937..ad47bb68798 100644
--- a/cmake/developer_package/cross_compile/python_helpers.cmake
+++ b/cmake/developer_package/cross_compile/python_helpers.cmake
@@ -23,7 +23,7 @@ function(ov_detect_python_module_extension)
     elseif(AARCH64)
         set(python3_config aarch64-linux-gnu-python3-config)
     elseif(X86_64)
-        set(python3_config x86_64-linux-gnu-python3-config)
+        set(python3_config python3-config)
     else()
         message(WARNING "Python cross-compilation warning: ${OV_ARCH} is unknown for python build. Please, specify PYTHON_MODULE_EXTENSION explicitly")
     endif()
diff --git a/cmake/developer_package/packaging/rpm/rpm.cmake b/cmake/developer_package/packaging/rpm/rpm.cmake
index 99f11730983..1a1f61fcd3d 100644
--- a/cmake/developer_package/packaging/rpm/rpm.cmake
+++ b/cmake/developer_package/packaging/rpm/rpm.cmake
@@ -156,7 +156,7 @@ ov_rpm_specific_settings()
 # needed to add triggers for packages with libraries
 set(def_triggers "${OpenVINO_BINARY_DIR}/_CPack_Packages/triggers")
 set(triggers_content "# /bin/sh -p\n/sbin/ldconfig\n")
-file(WRITE "${def_triggers}" "${triggers_content}")
+#file(WRITE "${def_triggers}" "${triggers_content}")
 
 #
 # Functions helpful for packaging your modules with RPM cpack
diff --git a/cmake/developer_package/target_flags.cmake b/cmake/developer_package/target_flags.cmake
index c6e3ebb2906..5891a1397e6 100644
--- a/cmake/developer_package/target_flags.cmake
+++ b/cmake/developer_package/target_flags.cmake
@@ -207,7 +207,7 @@ function(ov_libc_version)
     endif()
 endfunction()
 
-ov_libc_version()
+#ov_libc_version()
 
 #
 # Detects default value for _GLIBCXX_USE_CXX11_ABI for current compiler
@@ -218,4 +218,4 @@ macro(ov_get_glibcxx_use_cxx11_abi)
     endif()
 endmacro()
 
-ov_get_glibcxx_use_cxx11_abi()
+#ov_get_glibcxx_use_cxx11_abi()
diff --git a/samples/cpp/CMakeLists.txt b/samples/cpp/CMakeLists.txt
index 695dbbe7955..612c6e383c1 100644
--- a/samples/cpp/CMakeLists.txt
+++ b/samples/cpp/CMakeLists.txt
@@ -206,9 +206,9 @@ macro(ov_add_sample)
     target_link_libraries(${SAMPLE_NAME} PRIVATE ${ov_link_libraries} Threads::Threads ${SAMPLE_DEPENDENCIES})
 
     install(TARGETS ${SAMPLE_NAME}
-            RUNTIME DESTINATION samples_bin/
-            COMPONENT samples_bin
-            EXCLUDE_FROM_ALL)
+            DESTINATION ${CMAKE_INSTALL_BINDIR}
+            COMPONENT samples_bin)
+
 
     # create global target with all samples / demo apps
     if(NOT TARGET ov_samples)
diff --git a/src/bindings/python/CMakeLists.txt b/src/bindings/python/CMakeLists.txt
index f075dc21250..035963e6bc2 100644
--- a/src/bindings/python/CMakeLists.txt
+++ b/src/bindings/python/CMakeLists.txt
@@ -324,7 +324,7 @@ if(ENABLE_PYTHON_PACKAGING)
     # install OpenVINO Python API
 
     set(ov_python_package_prefix "${CMAKE_CURRENT_BINARY_DIR}/openvino/install_${pyversion}")
-    set(ov_install_lib "${ov_python_package_prefix}/lib/${python_versioned_folder}/${ov_site_packages}")
+    set(ov_install_lib "${ov_python_package_prefix}/${CMAKE_INSTALL_LIBDIR}/${python_versioned_folder}/${ov_site_packages}")
     set(openvino_meta_info_subdir "openvino-${OpenVINO_VERSION}-py${python_xy}.egg-info")
     set(openvino_meta_info_file "${ov_install_lib}/${openvino_meta_info_subdir}/PKG-INFO")
 
-- 
2.34.1

